### PR TITLE
Add SBOM logging to scribe.Emitter

### PR DESCRIPTION
**Pull request by fg-j**
## Summary

Now that Paketo buildpacks are beginning to generating SBOM files, it's useful to have consistent logging. 

Note: Currently, the `FormattingSBOM` function is a debug-only logger. This is the first time a debug-only logging function has been introduced in the `scribe.Emitter`.

Another option could be to create a `LeveledEmitter` struct  in the way that we have a `LeveledLogger` one. With that change, a buildpack author could do something like
```
emitter.Debug.FormattingSBOM(dependency)
```

I've been playing with that implementation – it adds complexity to the emitter, which biases me against the change. I think it's ok for the emitter to be opinionated about whether lines are logged at the INFO or DEBUG level. However, open to others' thoughts on this.

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by ForestEckhardt** [comment]
@paketo-buildpacks/tooling-maintainers I approve of this PR. I am going to hold off on formal approval until more eyes have looked at this as the introduction of a DEBUG only emitter sets an interesting precedent.

**Comment by sophiewigmore** [comment]
I don't mind the introduction of a debug-only log function, even if it is a departure from how we do things now. It enforces that this can really only be used in `DEBUG` mode, leading to alignment of SBOM logging practices across buildpacks for free.

This doesn't break the guarantees made in [the log level RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0027-log-levels.md) in any way either.

**Review by sophiewigmore — APPROVED** [review]

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**
